src/Entity/ReactionRuleConfig.php

```diff
@@ -121,11 +121,16 @@ class ReactionRuleConfig extends ConfigEntityBase {
   protected $module = 'rules';
 
   /**
-   * The event name this reaction rule is reacting on.
+   * The events this reaction rule is reacting on.
    *
-   * @var string
+   * Events array. The array is numerically indexed and contains arrays with the
+   * following structure:
+   *   - event_name: String with the event machine name.
+   *   - configuration: An array containing the event configuration.
```

Comment:
Agreed, but can we do that with CMI? I've only seen it save/load arrays.
As alternative, we could use an order array with
>$event_name => $configuration
which Alex tried initially. Problem there was that we need to have dots in event names (as symfony events use dots), but CMI does not allow dots in config keys. Thus, we'd have to massage the event names and somehow escape/replace dots. That makes it more complicated, also when querying for events etc.